python/dgl/graphbolt/impl/ondisk_dataset.py

```diff
@@ -46,168 +59,181 @@ def preprocess_ondisk_dataset(input_config_path: str) -> str:
     output_config_path : str
         The path to the output config file.
     """
-    # 0. Load the input_config.
-    with open(input_config_path, "r") as f:
-        input_config = yaml.safe_load(f)
-
-    # If the input config does not contain the "graph" field, then we
-    # assume that the input config is already preprocessed.
-    if "graph" not in input_config:
-        print("The input config is already preprocessed.")
-        return input_config_path
-
-    print("Start to preprocess the on-disk dataset.")
-    # Infer the dataset path from the input config path.
-    dataset_path = Path(os.path.dirname(input_config_path))
-    processed_dir_prefix = Path("preprocessed")
-
-    # 1. Make `processed_dir_prefix` directory if it does not exist.
-    os.makedirs(dataset_path / processed_dir_prefix, exist_ok=True)
-    output_config = deepcopy(input_config)
-
-    # 2. Load the edge data and create a DGLGraph.
-    is_homogeneous = "type" not in input_config["graph"]["nodes"][0]
-    if is_homogeneous:
-        # Homogeneous graph.
-        num_nodes = input_config["graph"]["nodes"][0]["num"]
-        edge_data = pd.read_csv(
-            dataset_path / input_config["graph"]["edges"][0]["path"],
-            names=["src", "dst"],
-        )
-        src, dst = edge_data["src"].to_numpy(), edge_data["dst"].to_numpy()
-
-        g = dgl.graph((src, dst), num_nodes=num_nodes)
-    else:
-        # Heterogeneous graph.
-        # Construct the num nodes dict.
-        num_nodes_dict = {}
-        for node_info in input_config["graph"]["nodes"]:
-            num_nodes_dict[node_info["type"]] = node_info["num"]
-        # Construct the data dict.
-        data_dict = {}
-        for edge_info in input_config["graph"]["edges"]:
+    # Check if the dataset path is valid.
+    if not os.path.exists(dataset_path):
+        raise RuntimeError(f"Invalid dataset path: {dataset_path}")
+
+    # Fix all paths under dataset_path.
+    with fix_dataset_path(dataset_path):
+        # 0. Check if the dataset is already preprocessed.
```

Comment:
this `with` body is so large, is it possible to split these steps into separate methods?

Reply:
+1

Reply:
One question, Do we need to directly modify the `output_config` in sub-functions？

Since we need to modify the `output_config` in each stage, the question arises whether the modification should take place in the `preprocess` function or in the sub-functions. Making these modifications within the sub-functions might seem a bit unconventional.